### PR TITLE
FUSETOOLS2-428 - added sql extension to supported file types for the lsp

### PR DESCRIPTION
**Pull request by lhein**
Signed-off-by: Lars Heinemann <lhein.smx@gmail.com>

### CODE REVIEW

File: src/languageServer.ts [hunk 1]
```diff
@@ -32,11 +32,12 @@ export function activate(context: ExtensionContext) {
 	};
 
 	let clientOptions: LanguageClientOptions = {
-		documentSelector: ['ddl'],
+		documentSelector: ['ddl', 'sql'],
 		synchronize: {
 			configurationSection: ['ddl'],
```
Comment (bfitzpat) [inline src/languageServer.ts:37]: Does this also have to change?

Comment (lhein) [reply src/languageServer.ts:37]: No. If I do that then it will override the documentselector for the sql extension and that extension then stops working.

Comment (bfitzpat) [reply src/languageServer.ts:37]: So what's the difference between the configurationSection and the documentSelector in this case? Are we just sharing the configuration between two file types now?

Comment (lhein) [reply src/languageServer.ts:37]: documentSelector registers this LSP for a given document type / language. The sql language is defined outside of our extension and just reused if available. 
For configurationSection I am a bit puzzled. It seems to be some kind of vscode config reference but I did not find useful information yet. The only thing I can tell you is that once I put in 'sql' then the sql extension does no longer work.

Comment (bfitzpat) [reply src/languageServer.ts:37]: Looks like maybe it's about the settings based on this comment - https://github.com/microsoft/vscode-languageserver-node-example/blob/1fd1ee0d52151564ac16a8ffdbe736492223624d/client/src/extension.ts#L32

Comment (bfitzpat) [reply src/languageServer.ts:37]: There's a little more here - https://vscode-docs.readthedocs.io/en/stable/extensions/example-language-server/

Comment (lhein) [reply src/languageServer.ts:37]: Doesn't sound like I want to do that for sql.

Comment (lhein) [reply src/languageServer.ts:37]: btw just as a side note:

image

Comment (bfitzpat) [reply src/languageServer.ts:37]: Seems we may want to spin this configurationSection bit off into its own issue and deal with it separately? WDYT @apupier ?

Comment (apupier) [reply src/languageServer.ts:37]: agree

**Comment by lhein** [comment]
image

**Comment by lhein** [comment]
sql_coloring_and_completion

**Comment by lhein** [comment]
Now that new test will fail as we need a new Teiid LSP version.

**Comment by lhein** [comment]
Locally it works fine with the modified LSP:

image

**Comment by apupier** [comment]
in the video you are showing the MySQl syntax extension to be installed. So you mean it is a requirement? So maybe it needs to be indicated as dependency of the plugin.
Another way should be to provide the sql language configuration too (but check that i doesn't cause troubles when both are defined). So it will work even without the extension installed.

If MySQl syntax is chosen as dependency, to have the test working, it needs ot be installed  in test preparation. Something similar to https://github.com/camel-tooling/vscode-camelk/blob/e240b872c1d096bf140cf68f4ae7490ddd071dfb/src/test/runTests.ts#L16-L18

**Comment by lhein** [comment]
> in the video you are showing the MySQl syntax extension to be installed. So you mean it is a requirement?

No, it won't be a requirement but at some point we will create an extension pack which then includes this one for better syntax coloring.

**Comment by lhein** [comment]
Something is definitely not working as expected. I uninstalled the MySQL extension and it seems that breaks the code completion for me.

**Pull request merged**